RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/RecoverWalletWithoutProfile/Coordinator/RecoverWalletWithoutProfileCoordinator.swift

```diff
@@ -118,30 +116,45 @@ public struct RecoverWalletWithoutProfileCoordinator: Sendable, FeatureReducer {
 		case .path(.element(_, action: .recoverWalletControlWithBDFSOnly(.delegate(.continue)))):
 			state.path.append(
 				.importMnemonic(
-					.init(
-						// We SHOULD remove the mnemonic from keychain if it we do not
-						// complete this flow.
-						persistStrategy: .init(
-							mnemonicForFactorSourceKind: .onDevice(
-								.babylon
-							),
-							location: .intoKeychainOnly
-						)
-					)
+					.init(header: .init(title: L10n.EnterSeedPhrase.Header.title), persistStrategy: nil, wordCount: .twentyFour)
 				)
 			)
 			return .none
 
 		case let .path(.element(_, action: .importMnemonic(.delegate(delegateAction)))):
 			switch delegateAction {
-			case let .persistedMnemonicInKeychainOnly(factorSource):
-				guard let fromHash = factorSource.id.extract(FactorSource.ID.FromHash.self) else {
-					fatalError("error handling")
+			case let .notPersisted(mnemonicWithPassphrase):
+				do {
+					let fromHash = try FactorSource.id(
+						fromMnemonicWithPassphrase: mnemonicWithPassphrase,
+						factorSourceKind: .device
+					)
+
+					let deviceFactorSource = DeviceFactorSource(
+						id: fromHash,
+						common: .init(),
+						hint: .init(
+							name: "iPhone",
```

Comment:
gets replaced with `device` dependencies `await device.model` and `await device.name` inside of `ProfileStore` before being saved into keychain.